tsp-output/@azure-tools/typespec-csharp/src/Generated/Audio.cs

```diff
@@ -344,92 +336,90 @@ public virtual async Task<Response> CreateTranslationAsync(RequestContent conten
-            }
-            catch (Exception e)
+            options ??= new RequestOptions();
+
+            using PipelineMessage message = CreateCreateTranslationRequest(content, contentType, options);
+
+            _pipeline.Send(message);
+
+            PipelineResponse response = message.Response!;
+
+            if (response.IsError && options.ErrorOptions == ClientErrorBehaviors.Default)
             {
-                scope.Failed(e);
-                throw;
+                throw new ClientResultException(response);
             }
+
+            return ClientResult.FromResponse(response);
         }
 
-        internal HttpMessage CreateCreateSpeechRequest(RequestContent content, RequestContext context)
+        internal PipelineMessage CreateCreateSpeechRequest(BinaryContent content, RequestOptions options)
         {
-            var message = _pipeline.CreateMessage(context, ResponseClassifier200);
+            var message = _pipeline.CreateMessage();
+            message.ResponseClassifier = ResponseErrorClassifier200;
             var request = message.Request;
-            request.Method = RequestMethod.Post;
-            var uri = new RawRequestUriBuilder();
+            request.Method = "POST";
+            var uri = new ClientUriBuilder();
             uri.Reset(_endpoint);
             uri.AppendPath("/audio/speech", false);
-            request.Uri = uri;
-            request.Headers.Add("Accept", "application/octet-stream");
-            request.Headers.Add("Content-Type", "application/json");
+            request.Uri = uri.ToUri();
+            request.Headers.Set("Accept", "application/octet-stream");
+            request.Headers.Set("Content-Type", "application/json");
             request.Content = content;
+            message.Apply(options);
             return message;
         }
 
-        internal HttpMessage CreateCreateTranscriptionRequest(RequestContent content, RequestContext context)
+        internal PipelineMessage CreateCreateTranscriptionRequest(BinaryContent content, string contentType, RequestOptions options)
         {
-            var message = _pipeline.CreateMessage(context, ResponseClassifier200);
+            var message = _pipeline.CreateMessage();
+            message.ResponseClassifier = ResponseErrorClassifier200;
             var request = message.Request;
-            request.Method = RequestMethod.Post;
-            var uri = new RawRequestUriBuilder();
+            request.Method = "POST";
+            var uri = new ClientUriBuilder();
             uri.Reset(_endpoint);
             uri.AppendPath("/audio/transcriptions", false);
-            request.Uri = uri;
-            request.Headers.Add("Accept", "application/json");
-            request.Headers.Add("content-type", "multipart/form-data");
+            request.Uri = uri.ToUri();
+            request.Headers.Set("Accept", "application/json");
+            request.Headers.Set("Content-Type", contentType);
```

Comment:
multipart/form-data request creation helpers will set the `Content-Type` header from the value passed to the protocol method.